dotcom-rendering/src/lib/article.ts

```diff
@@ -15,40 +14,39 @@ const enhancePinnedPost = (
 	renderingTarget: RenderingTarget,
 	block?: Block,
 ) => {
-	return block ? enhanceBlocks([block], format, renderingTarget)[0] : block;
+	if (!block) return;
+	return enhanceBlocks([block], format, {
+		renderingTarget,
+		imagesForLightbox: [],
+		promotedNewsletter: undefined,
+	})[0];
 };
 
 export const enhanceArticleType = (
 	body: unknown,
 	renderingTarget: RenderingTarget,
 ): DCRArticle => {
-	const validated = validateAsArticleType(body);
-	// addImageIDs needs to take account of both main media elements
-	// and block elements, so it needs to be executed here
-	const { mainMediaElements, blocks } = addImageIDs(validated);
-	const data = {
-		...validated,
-		mainMediaElements,
-		blocks,
-	};
+	const data = validateAsArticleType(body);
 
-	const enhancedBlocks = enhanceBlocks(
-		data.blocks,
-		data.format,
+	const imagesForLightbox = data.config.switches.lightbox
+		? buildLightboxImages(data.format, data.blocks, data.mainMediaElements)
+		: [];
```

Comment:
Now we start by building the images for lightbox, and then optionally add the relevant positions to main media and block image elements if there’s a match.